eng/Tools.props

```diff
@@ -61,6 +63,36 @@
     <OnError ExecuteTargets="DumpInitToolsLog" />
   </Target>
 
+  <Target Name="InitGlobalTools" AfterTargets="Restore">
+
+    <ItemGroup>
+      <RepoTool Include="coverlet.console" Version="$(CoverletConsolePackageVersion)" />
+      <RepoTool Include="dotnet-reportgenerator-globaltool" Version="$(DotNetReportGeneratorGlobalToolPackageVersion)" />
+    </ItemGroup>
+
+    <PropertyGroup>
+      <RepoToolIdentifierEscaped>$([System.Text.RegularExpressions.Regex]::Escape('%(RepoTool.Identity)'))</RepoToolIdentifierEscaped>
```

Comment:
I took a look at what dotnet install / dotnet list does and it seems to create a `.store` folder with packageId\version underneath.  I wonder if we could just probe for this rather than doing a list and parsing.  Even if just as a stand-in until we can get something better from SDK.

Reply:
I use batching in the escape string and it works as expected. Am I missing something?

Reply:
I'll take a second look, maybe I'm really missing something here.

Reply:
@ericstj I couldn't find any issues here (see screenshot)

![image](https://user-images.githubusercontent.com/7412651/49480455-0d93c080-f827-11e8-9acb-bf570efe3c46.png)

Reply:
Thanks, the intermediate property was indeed an issue and resulted in the wrong behavior. I refined that now and got rid of the properties. The advantage with my solution is that dotnet tool list will only be executed once and we don't rely on internal behaviors on dotnet tool (Eric's suggestion). The downside is that got very long but I can live with that.